packages/data/lib/database/dao/movie_character_dao.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:data/database/entities/movie_character.dart';
+import 'package:floor/floor.dart';
+
+@dao
+abstract class MovieCharacterDao {
+  @Query('SELECT * FROM MovieCharacter WHERE movieId = :movieId')
+  Future<List<MovieCharacter>> findMovieCast(int movieId);
+
+  @Query('DELETE FROM MovieCharacter WHERE movieId NOT IN (:movieIds)')
+  Future<void> deleteCastsExceptWithIds(List<int> movieIds);
```

Comment:
u shouldn't have the separate query for deleting cast. it should be deleted by cascade when u delete the movie. but u need to link the tables

Reply:
Fixed.